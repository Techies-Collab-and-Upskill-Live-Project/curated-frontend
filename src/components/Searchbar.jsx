// import { IconSearch } from '@tabler/icons-react';

export default function Searchbar() {
  return (
    <div className="w-full max-w-2xl relative">
      <div className="flex items-center border rounded-[10px] gap-2 mb-2">
        {/* Input Container with Icon */}
        <div className="relative flex-1">
          <IconSearch className="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-900" />
          <input
            type="text"
            placeholder="Search/Playlist Link..."
            className="w-full pl-10 pr-4 py-2 focus:outline-none bg-transparent"
          />
        </div>
        
        {/* Search Button */}
        <button className="bg-primary text-white px-6 py-2 rounded-md transition-colors">
          Search
        </button>
      </div>
    </div>
  );
}
