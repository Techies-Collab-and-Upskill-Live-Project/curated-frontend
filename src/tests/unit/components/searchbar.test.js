import { render, screen, fireEvent } from '@testing-library/react';
import Searchbar from '@/components/Searchbar'; 
import { IconSearch } from '@tabler/icons-react';

jest.mock('@tabler/icons-react', () => ({
  IconSearch: jest.fn(() => <svg data-testid="search-icon" />),
}));

describe('Searchbar Component', () => {
  const mockOnSearch = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    render(<Searchbar onSearch={mockOnSearch} />);
  });

  it('renders the search form with all elements', () => {
    expect(screen.getByTestId('search-form')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Search/Playlist Link...')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /search/i })).toBeInTheDocument();
    expect(screen.getByTestId('search-icon')).toBeInTheDocument();
  });

  it('updates the query state when typing', () => {
    const input = screen.getByPlaceholderText('Search/Playlist Link...');
    fireEvent.change(input, { target: { value: 'test query' } });
    expect(input.value).toBe('test query');
  });

  it('calls onSearch with the query when form is submitted', () => {
    const input = screen.getByPlaceholderText('Search/Playlist Link...');
    const form = screen.getByTestId('search-form');

    fireEvent.change(input, { target: { value: 'react testing' } });
    fireEvent.submit(form);

    expect(mockOnSearch).toHaveBeenCalledTimes(1);
    expect(mockOnSearch).toHaveBeenCalledWith('react testing');
  });

  it('prevents default form submission behavior', () => {
    const form = screen.getByTestId('search-form');
    const preventDefault = jest.fn();
    const submitEvent = new Event('submit', { bubbles: true });
    submitEvent.preventDefault = preventDefault;
    
    fireEvent(form, submitEvent);
    expect(preventDefault).toHaveBeenCalledTimes(1);
  });

  it('requires the input field to be filled', () => {
    const input = screen.getByPlaceholderText('Search/Playlist Link...');
    expect(input).toBeRequired();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Searchbar onSearch={mockOnSearch} />);
    expect(asFragment()).toMatchSnapshot();
  });
});