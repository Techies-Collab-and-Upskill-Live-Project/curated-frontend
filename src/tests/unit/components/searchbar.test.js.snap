// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Searchbar Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full px-4 sm:px-0 max-w-2xl relative"
  >
    <form
      class="flex items-center text-white border rounded-[10px] gap-1 sm:gap-2 mb-2"
    >
      <div
        class="relative flex-1"
      >
        <svg
          data-testid="search-icon"
        />
        <input
          class="w-full pl-8 sm:pl-10 pr-2 sm:pr-4 py-2 text-sm sm:text-base focus:outline-none bg-transparent"
          placeholder="Search/Playlist Link..."
          required=""
          type="text"
          value=""
        />
      </div>
      <button
        class="bg-primary text-white px-3 sm:px-6 py-2 text-sm sm:text-base rounded-md transition-colors"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
